### ISSUE TITLE
Support for class methods

### ISSUE THREAD

**Issue by LLazarek** [LLazarek/sit#1]
Sit fails to provide support for testing class methods; it generates an empty test file.

**Comment by Captain456** [comment]
We could probably fix this by adding a special test-class keyword. You would provide the class type and it would create an instantiation of the class with a generic name and test the function.

This could be expanded to functions that mutate a data member, etc.

**Comment by LLazarek** [comment]
I agree; providing the class name seemed redundant at first but would be useful in the case of testing functions defined inline with the class definition.
I think that this keyword should be nearly identical to test-input; the only difference should be that we add an object instantiation along with the parameter instantiations. We can handle the problem of classes without default constructors by doing the same deferred declarations strategy we used for references.

The only issue I see with this is that our syntax is becoming rather cumbersome for performing multiple tests: every test has to provide setup code to create an object and manipulate it before the function is run. Perhaps we should provide a feature for providing a single set of setup code to be executed prior to a series of tests?
